build.rs

```diff
@@ -25,11 +37,45 @@ fn main() {
     // purpose. `OUT_DIR` points to a directory within target, whose path even
     // contains a hash. This configuration file needs to be referenced from the
     // GDB configuration, which can't just ask Cargo where to look for it.
+    fs::create_dir_all("target")?;
+    File::create("target/openocd.cfg")?.write_all(openocd_cfg)?;
```

Comment:
If the errors are just bubbled up, why not just `.unwrap()`?

Reply:
I think this reads more nicely. `unwrap` and `expect` breaks the flow of what's actually supposed to happen.

In general, I believe that lower layers should report errors, and leave it to upper layers to handle them. But granted, that doesn't make much of a difference in this tiny script.